include/deal.II/fe/mapping_fe_field.h

```diff
@@ -649,10 +660,57 @@ class MappingFEField : public Mapping<dim, spacedim>
 
 
   // Declare other MappingFEField classes friends.
-  template <int, int, class, class>
+  template <int, int, class, class, class>
   friend class MappingFEField;
 };
 
+
+
+/**
+ * The MappingFEField is a generalization of the MappingQEulerian class, for
+ * arbitrary vector finite elements. The two main differences are that this
+ * class uses a vector of absolute positions instead of a vector of
+ * displacements, and it allows for arbitrary FiniteElement types, instead of
+ * only FE_Q.
+ *
+ * This class effectively decouples the topology from the geometry, by
+ * relegating all geometrical information to some components of a
+ * FiniteElement vector field. The components that are used for the geometry
+ * can be arbitrarily selected at construction time.
+ *
+ * The idea is to consider the Triangulation as a parameter configuration
+ * space, on which we construct an arbitrary geometrical mapping, using the
+ * instruments of the deal.II library: a vector of degrees of freedom, a
+ * DoFHandler associated to the geometry of the problem and a ComponentMask
+ * that tells us which components of the FiniteElement to use for the mapping.
+ *
+ * Typically, the DoFHandler operates on a finite element that is constructed
+ * as a system element (FESystem()) from continuous FE_Q() (for iso-parametric
+ * discretizations) or FE_Bernstein() (for iso-geometric discretizations)
+ * objects. An example is shown below:
+ *
+ * @code
+ *    const FE_Q<dim,spacedim> feq(1);
+ *    const FESystem<dim,spacedim> fesystem(feq, spacedim);
+ *    DoFHandler<dim,spacedim> dhq(triangulation);
+ *    dhq.distribute_dofs(fesystem);
+ *    const ComponentMask mask(spacedim, true);
+ *    Vector<double> eulerq(dhq.n_dofs());
+ *    // Fills the euler vector with information from the Triangulation
+ *    VectorTools::get_position_vector(dhq, eulerq, mask);
+ *    MappingFEField<dim,spacedim> map(dhq, eulerq, mask);
+ * @endcode
```

Comment:
Given that this class is now deprecated, can this comment be deleted?

Reply:
I wanted to use `@copydoc` to grab the documentation from the non-deprecated class, but `doxygen` can not differentiate between specializations by their template argument :-(

I don't know what the general consent on the documentation of deprecated classes and functions is. I've seen some with and without a separate documentation, depending on whether the deprecated functionality differs from the one ought to be kept.

Considering that this is an exact replica, we should be fine with removing it and just leaving the link to the documentation with the `@deprecated` flag. What do you think?

Reply:
Yes, that would be exactly what I had in mind.